api/app.go

```diff
@@ -1692,7 +1692,11 @@ func registerUnit(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	}
 	err = a.RegisterUnit(hostname, customData)
 	if err != nil {
-		if _, ok := err.(*provision.UnitNotFoundError); ok {
+		if err, ok := err.(*provision.UnitNotFoundError); ok {
+			if r.Header.Get("x-Agent-Version") == "" {
+				msgError := fmt.Sprintf("Please contact admin. %s platform is using outdated deploy-agent version, minimum required version is 0.2.4", a.GetPlatform())
+				return &errors.HTTP{Code: http.StatusUpgradeRequired, Message: msgError}
```

Comment:
i would just return `http.StatusBadRequest` here as `http.StatusUpgradeRequired` actually means that the protocol version is not supported